### ISSUE TITLE
Deleting files produces empty commits

### ISSUE THREAD

**Issue by klieret** [lindell/multi-gitter#266]
**Describe the bug**
Deleting a file produces empty commits instead of deleting them

**To Reproduce**

Run the following command on a test repository containing a `README.md` (you can fork [mine](https://github.com/klieret/multi-gitter-delete-bug-demonstration)):

```bash
multi-gitter run "rm README.md" -R klieret/multi-gitter-delete-bug-demonstration -m "Remove file"
```

**Expected behavior**

I expect a commit that has `README.md` deleted.

**What happens instead**

The output looks normal:

```
INFO[0000] Running on 1 repositories
INFO[0000] Cloning and running script                    repo=klieret/multi-gitter-delete-bug-demonstration
INFO[0000] Pushing changes to remote                     repo=klieret/multi-gitter-delete-bug-demonstration
INFO[0001] Creating pull request                         repo=klieret/multi-gitter-delete-bug-demonstration
Repositories with a successful run:
  klieret/multi-gitter-delete-bug-demonstration #1
```

But you get a [PR with an empty commit](https://github.com/klieret/multi-gitter-delete-bug-demonstration/pull/1). Which is bad because the `README.md` should be deleted, but also bad because empty commits shouldn't be added by multi-gitter.

**Comment by lindell** [comment]
Thanks for a concise repo :D This does indeed seems like a bug. I'm unable to repo, or fix right now as I don't have access to a dev computer. But can you try to run with --git-type=cmd and see if it gives the same result?

**Comment by klieret** [comment]
Will try later!

I was also wondering whether maybe using `git rm` instead of plain `rm` might solve this? (or a manual `git add` afterwards)

**Comment by lindell** [comment]
Running git commands should not be needed, and could cause strange behavior.

**Comment by klieret** [comment]
Indeed `git-type=cmd` solves this. So it's an issue with go-git?

**Comment by lindell** [comment]
Could be, or more likely how multi-gitter is using it. But one of the two guaranteed ☺️

Can't see anything obvious here: https://github.com/lindell/multi-gitter/blob/5dfd6d9fc877d06f905cbeb27e39305d16afee65/internal/git/gogit/git.go#L91
But will have to look closer when I have access to a computer:)

**Comment by lindell** [comment]
Seems that this is actually a go-git issue: https://github.com/go-git/go-git/issues/223

There is a linked workaround that I will test later this week to see if it works (or if you want to try it out).

**Comment by lindell** [comment]
Added a fix here: #273 with a test that fails without the fix.
